### ISSUE TITLE
scale subresource not accepted on stable crd derives

### ISSUE THREAD

**Issue by clux** [clux/kube-rs#387]
A follow up to the mysterious #176.

With the following diff with the `crd_apply` example:

```diff
diff --git examples/crd_apply.rs examples/crd_apply.rs
index f9ce9b33..95bdbbc0 100644
--- examples/crd_apply.rs
+++ examples/crd_apply.rs
@@ -18,7 +18,7 @@ use kube::{
 #[derive(CustomResource, Deserialize, Serialize, Clone, Debug, JsonSchema)]
 #[kube(group = "clux.dev", version = "v1", kind = "Foo", namespaced)]
 #[kube(status = "FooStatus")]
-//#[kube(scale = r#"{"specReplicasPath":".spec.replicas", "statusReplicasPath":".status.replicas"}"#)]
+#[kube(scale = r#"{"specReplicasPath":".spec.replicas", "statusReplicasPath":".status.replicas"}"#)]
 pub struct FooSpec {
     name: String,
     info: Option<String>,

```

I.e.: if using `#[kube(scale = r#"{"specReplicasPath":".spec.replicas", "statusReplicasPath":".status.replicas"}"#)]` on a custom resource derive, whatever handles the status subresource data fails to accept our serialization for `.spec.replicas` (no matter what we put it to).

Here's a patch on a foo with integer serialization:

```
[2021-01-16T23:33:15Z INFO  crd_apply] Applying 1: 
    ---
    apiVersion: clux.dev/v1
    kind: Foo
    metadata:
      name: baz
    spec:
      name: baz
      info: old baz
      replicas: 3
Error: ApiError: Foo.clux.dev "baz" is invalid: .spec.replicas: Invalid value: 0: .spec.replicas accessor error: 3 is of the type float64, expected int64: Invalid (ErrorResponse { status: "Failure", message: "Foo.clux.dev \"baz\" is invalid: .spec.replicas: Invalid value: 0: .spec.replicas accessor error: 3 is of the type float64, expected int64", reason: "Invalid", code: 422 })

Caused by:
    Foo.clux.dev "baz" is invalid: .spec.replicas: Invalid value: 0: .spec.replicas accessor error: 3 is of the type float64, expected int64: Invalid
```

A nonsense error. The .spec.replicas value we serialized is not a float. In fact any type we set the various replicas field to does not change the error.
k8s also accepts the value without the scale subresource instruction, so it must be something wrong related to that, but haven't gotten to the bottom of it.

**Comment by clux** [comment]
Looks like an instance of https://github.com/kubernetes/kubernetes/issues/95680.

**Comment by clux** [comment]
It's a kubernetes bug for sure, fixed by upgrading from 1.19.4 to 1.20.2.
